pkgs/by-name/ni/nixfmt-tree/package.nix

```diff
@@ -1,45 +1,63 @@
 {
   lib,
   runCommand,
-  buildEnv,
-  makeWrapper,
-  writers,
   treefmt,
   nixfmt-rfc-style,
   nixfmt-tree,
 
-  settings ? {
-    # The default is warn, which would be too annoying for people who just care about Nix
-    on-unmatched = "info";
-    # Assumes the user is using Git, fails if it's not
-    tree-root-file = ".git/index";
-
-    formatter.nixfmt = {
-      command = "nixfmt";
-      includes = [ "*.nix" ];
-    };
-  },
-  runtimePackages ? [
+  settings ? { },
+  runtimeInputs ? [
     nixfmt-rfc-style
   ],
-}:
-buildEnv {
-  name = "nixfmt-tree";
-
-  # Allows this derivation to be used as a shell providing both treefmt and nixfmt
-  paths = [ treefmt ] ++ runtimePackages;
-  pathsToLink = [ "/bin" ];
-
-  nativeBuildInputs = [
-    makeWrapper
-  ];
-
-  postBuild = ''
-    wrapProgram $out/bin/treefmt \
-      --prefix PATH : $out/bin \
-      --add-flags "--config-file ${writers.writeTOML "treefmt.toml" settings}"
-  '';
+  runtimePackages ? [ ],
+}@args:
+let
+  # Handle renamed argument (runtimePackages -> runtimeInputs) added 2025-03-19
+  runtimeInputs' =
+    if args ? runtimePackages then
+      lib.warn "nixfmt-tree: the `runtimePackages` argument has been renamed to `runtimeInputs` for consistency with `treefmt.withConfig`." runtimePackages
+    else
+      runtimeInputs;
+
+  # Check whether runtimeInputs has a nixfmt package
+  # Used below in warnings and conditional settings
+  isNixfmt = p: p.meta.mainProgram or null == "nixfmt";
+  hasNixfmt = lib.any isNixfmt runtimeInputs';
+
+  treefmtWithConfig = treefmt.withConfig {
+    name = "nixfmt-tree";
+
+    settings = [
+      # Default settings
+      {
+        _file = ./package.nix;
+
+        # Log level for files treefmt won't format
+        # The default is warn, which would be too annoying for people who just care about Nix
+        on-unmatched = lib.mkOptionDefault "info";
```

Comment:
Not your fault, but hasn't this changed in recent versions of treefmt? https://github.com/numtide/treefmt/issues/496

Reply:
Looking at the docs, I believe the default is currently "warn".

I assume "info" is also visible at the default log level, but is probably formatted less obnoxiously than warnings?